pedalboard/io/PythonInputStream.h

```diff
@@ -219,4 +234,63 @@ class PythonInputStream : public juce::InputStream, public PythonFileLike {
   juce::int64 totalLength = -1;
   bool lastReadWasSmallerThanExpected = false;
 };
+
+/**
+ * A juce::InputStream subclass that fetches its
+ * data from a provided Python file-like object.
+ */
+class PythonMemoryViewInputStream : public PythonInputStream {
+public:
+  PythonMemoryViewInputStream(py::buffer bufferLike, py::object passedObject)
+      : PythonInputStream(bufferLike) {
+    info = bufferLike.request();
+    totalLength = info.size * info.itemsize;
+    repr = py::repr(passedObject).cast<std::string>();
+
+    if (py::hasattr(passedObject, "tell")) {
+      try {
+        offset = std::min(
+            std::max(0LL, passedObject.attr("tell")().cast<juce::int64>()),
+            totalLength);
+      } catch (py::error_already_set e) {
+        e.restore();
+      } catch (std::exception e) {
+        // If we can't call tell(), that's fine; just assume an offset of 0.
+      }
+    }
+  }
+
+  virtual std::string getRepresentation() override { return repr; }
+
+  virtual bool isSeekable() noexcept override { return true; }
+
+  virtual juce::int64 getTotalLength() noexcept override { return totalLength; }
+
+  virtual int read(void *buffer, int bytesToRead) noexcept override {
+    int start = offset;
+    int end = std::min(offset + bytesToRead, totalLength);
```

Comment:
Nit: It may be easier to read if this is instead written as 
```c++ 
if (offset + bytesToRead >= totalLength) {
  bytesToRead = totalLength - offset;
}
```
Then the subtraction logic below doesn't need repeating and you don't need start. I sometimes get a little confused with `end` because even though `end` is usually exclusive, occasionally it isn't.